builds/e2e/connectivity.yaml

```diff
@@ -88,7 +88,7 @@ jobs:
               echo "run.steps=0"
             fi
       - checkout: none
-        condition: eq(variables['run.steps'], 1)
+        condition: and(succeeded(), eq(variables['run.steps'], 1))
```

Comment:
Per https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops,
you may want to be explicated that you want a strictly `Succeeded` status? 

By default, `succeeded()` is a combination of `Succeeded` and `SucceededWithIssues` (continued w/ error)

Reply:
As talked, succeeded() will ensure all previous steps are passed.  This is exactly what we want.